crates/rspack_core/src/normal_module.rs

```diff
@@ -466,7 +464,14 @@ impl NormalModule {
 impl Identifiable for NormalModule {
   #[inline]
   fn identifier(&self) -> ModuleIdentifier {
-    self.id
+    if !matches!(
+      self.module_type,
```

Comment:
As Rspack considers `.tsx|ts` files to be the first class citizens, you should keep the `module_type` in `ModuleIdentifier` for `ModuleType::Ts` and `ModuleType::Tsx`.